### DISCUSSION TITLE
Stop Permission Inheritance >> set unique permission .. is this possible using PnP Core SDK

### DISCUSSION THREAD

**Discussion by mvcsharepointdev** [pnp/pnpcore#1102] [Q&A] [answered]
Can we perfome the following actions using PnP Core SDK?

1) Access the permission of a folder

![image](https://user-images.githubusercontent.com/79726511/219803466-136c7210-85b2-4ca8-8b3d-75280cb284ae.png)

![image](https://user-images.githubusercontent.com/79726511/219803524-eadf5d74-09be-4445-975f-6b31b0e33254.png)

2) Stop Permission inheritance 

![image](https://user-images.githubusercontent.com/79726511/219803567-6b661e49-42f4-4be2-a585-3c49b93636d0.png)

3) Set unique permission

![image](https://user-images.githubusercontent.com/79726511/219803678-0861a2be-cab2-4b38-9ea5-b4e4c2ea7c3d.png)

please i do not want to share the folder using links >> i want to use the exact above steps? are those possible? Thanks

**Comment by MathijsVerbeeck** [comment, marked as answer]
This should be possible, yes.
You should be able to find the information that you require at the following destination: [https://pnp.github.io/pnpcore/using-the-sdk/security-intro.html#assigning-roles](https://pnp.github.io/pnpcore/using-the-sdk/security-intro.html#assigning-roles)

**Reply by jansenbe** [reply]
This should indeed be possible. Thanks @MathijsVerbeeck
